dali/test/python/test_dali_variable_batch_size.py

```diff
@@ -98,13 +124,22 @@ def generate_data(max_batch_size, n_iter, sample_shape, lo=0., hi=1., dtype=np.f
 
 
 def single_op_pipeline(max_batch_size, input_data, device, *, input_layout=None,
-                       operator_fn=None, **opfn_args):
+                       operator_fn=None, needs_input=True, **opfn_args):
     pipe = Pipeline(batch_size=max_batch_size, num_threads=1, device_id=0)
     with pipe:
         input = fn.external_source(source=input_data, cycle=False, device=device,
                                    layout=input_layout)
-        output = input if operator_fn is None else operator_fn(input, device=device, **opfn_args)
-        pipe.set_outputs(output)
+        if operator_fn is None:
+            output = input
+        else:
+            if needs_input:
+                output = operator_fn(input, **opfn_args)
+            else:
+                output = operator_fn(**opfn_args)
+        if needs_input:
+            pipe.set_outputs(output)
+        else:
+            pipe.set_outputs(output, input)
```

Comment:
This is a hack, so that the `external_source` won't be pruned, correct? If so, maybe a comment here would be good?